restli-tools/src/main/java/com/linkedin/restli/tools/symbol/RestLiSymbolTableProvider.java

```diff
@@ -220,13 +221,21 @@ public SymbolTable getSymbolTable(String symbolTableName)
   @Override
   public SymbolTable getRequestSymbolTable(URI requestUri)
   {
+    // If the URI prefix doesn't match, return null.
+    if (!requestUri.toString().startsWith(_uriPrefix))
+    {
+      return null;
+    }
```

Comment:
shouldn't requestUri will always start with uriPrefix?
reason for adding d2 check was for HTTP(s) prefixes adding /serviceName/resources or something else might be different from where symbol table endpoint is hosted since we are using zk serverNodeUri to construct symbol table API endpoint

Reply:
There is currently no guarantee that request URI has to start with uriPrefix. This check just ensures that.
Yes, practically it is going to be d2 or http(s) but the contract of this class itself makes no assumptions about that. Hence, its safer to code it this way.